### PR TITLE
Addresses PatchesJson6902 patches issue in MergeKustomizations function

### LINKED ISSUE #4136: Issue while merging PatchesJson6902 in MergeKustomizations
/kind bug

**What steps did you take and what happened:**
Added one more overlay for argo componet to update ClusterRoles using PatchesJson6902

content of argo kustomization(**argo/overlays/csf/kustomization.yaml**) as below 
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
bases:
- ../../base
patchesJson6902:
- target:
    group: rbac.authorization.k8s.io
    version: v1beta1
    kind: ClusterRole
    name: argo
  path: patch.yaml
- target:
    group: rbac.authorization.k8s.io
    version: v1beta1
    kind: ClusterRole
    name: argo-ui
  path: patch2.yaml
```
**patch.yaml &  patch2.yaml** contents are as below
- op: add
  path: /rules/0
  value:
    apiGroups:
      - extensions
    resourceNames:
      - privileged
    resources:
      - podsecuritypolicies
    verbs:
      - use

**What did you expect to happen:**
After kfctl generate I was expecting two patch files argo-ClusterRole.yaml &  argo-ui-ClusterRole.yaml will get created with proper patch link in  new kustomization.yaml
```
apiVersion: kustomize.config.k8s.io/v1beta1
bases:
- base
configurations:
- overlays/istio/params.yaml
kind: Kustomization
namespace: kubeflow
patchesJson6902:
- path: argo-ClusterRole.yaml
  target:
    group: rbac.authorization.k8s.io
    kind: ClusterRole
    name: argo
    version: v1beta1
- path: argo-ui-ClusterRole.yaml
  target:
    group: rbac.authorization.k8s.io
    kind: ClusterRole
    name: argo-ui
    version: v1beta1
resources:
- overlays/istio/virtual-service.yaml
```
Instead i see only argo-ui-ClusterRole.yaml file is created with both patches added which is wrong
```
ls kustomize/argo/
argo-ui-ClusterRole.yaml  base  kustomization.yaml  overlays
```

**Anything else you would like to add:**
I could see problem in this line https://github.com/kubeflow/kubeflow/blob/0d52bb563d5f2f28da0def25f9b55fa99eefb740/bootstrap/pkg/kfapp/kustomize/kustomize.go#L972 where differnt path is added  for each PatchesJson6902 key(name) patch but while creating new patch files only one file is created https://github.com/kubeflow/kubeflow/blob/0d52bb563d5f2f28da0def25f9b55fa99eefb740/bootstrap/pkg/kfapp/kustomize/kustomize.go#L993 

**Environment:**

- Kubeflow version: `build version v0.6.2`
- kfctl version: `kfctl v0.6.2-0-g47a0e4c7`
- Kubernetes platform: `k8s cluster`
- Kubernetes version: 
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:36:19Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
- OS :
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

### PR DESCRIPTION
**Pull request by hemantha-kumara**
Resolves #4136
Creating different patch file for each PatchesJson6902 ops instead of aggregatepatch

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4137)

### CODE REVIEW

**Comment by kkasravi** [comment]
@hemantha-kumara are there any test cases or unit tests you can add for this.
You could add a unit-test under kubeflow/manifests/tests or use the existing cases of 
PatchesJson6902

**Comment by hemantha-kumara** [comment]
> are there any test cases or unit tests you can add for this.
> You could add a unit-test under kubeflow/manifests/tests or use the existing cases of
> PatchesJson6902

Sure, i will re-push with testcase changes.

**Comment by kkasravi** [comment]
thanks @hemantha-kumara 

/ok-to-test

**Comment by kkasravi** [comment]
@hemantha-kumara could you rebase? I'll lgtm, approve as soon as you do
- git fetch upstream master
- git rebase -i upstream/master

I can add a test-case - i just ran into the same bug. 
Nice work!

**Comment by hemantha-kumara** [comment]
@kkasravi I have rebased and submitted.

**Comment by kkasravi** [comment]
thanks @hemantha-kumara - great work

/lgtm
/approve

**Comment by kkasravi** [comment]
@hemantha-kumara - some weirdness in the notebook test.
Maybe rebase from upstream ?
`git rebase -i upstream/master`

**Comment by hemantha-kumara** [comment]
> @hemantha-kumara - some weirdness in the notebook test.
> Maybe rebase from upstream ?
> `git rebase -i upstream/master`

@kkasravi
`git rebase -i upstream/master` was adding others commits also in my pull request, hence I tried rebase using below commands
```
git fetch upstream
git rebase upstream/master
```
then resolved conflicts manually and added kustomize.go file back and submitted using 
```
git rebase --continue
git push origin MergeKustomizations -f
```

**Comment by kkasravi** [comment]
We should try and get this in for v0.7.0.

**Comment by hemantha-kumara** [comment]
as suggested created new PR #4225 and hence closing this PR

File: bootstrap/pkg/kfapp/kustomize/kustomize.go [hunk 1]
```diff
@@ -969,7 +969,6 @@ func MergeKustomizations(kfDef *kfdefsv3.KfDef, compDir string, overlayParams []
 			}
 		}
 		kustomization.PatchesJson6902 = make([]types.PatchJson6902, 0)
-		aggregatedPatchOps := make([]byte, 0)
```
[the remark this answers is not in the document]
Comment (hemantha-kumara) [reply bootstrap/pkg/kfapp/kustomize/kustomize.go]: aggregatedPatchOps  is no more required as creating new patch file after every target.
as suggested created new PR #4225 , pls have a look.

**Pull request closed without merging**